### ISSUE TITLE
Training scheme to choose

### ISSUE THREAD

**Issue by amardeepjaiman** [matterport/Mask_RCNN#1289]
Hi,

I am having around 40000 aerial images of size 500x500 to predict masks for buildings.Can anyone suggest is it necessary to train heads in this case ? is it fine if i only use 'all' layer option to fine tune all the layers. Currently the results I am getting after training 40 epochs of training 'heads' are not good. its leaving few building to generate mask and for most of them shapes are not good. Any other configuration parameter which can be adjusted ?

Thanks in advance,

**Comment by hackerghost93** [comment]
I don't think you should use fine tuning for this one. My guess would be training all the layers from the beginning and see the results. Your validation set will help you a lot in this one.

**Comment by amardeepjaiman** [comment]
Thanks @hackerghost93 for replying, so does it mean that , I dont need to train the heads and I can train all the layers using below option ? isnt it ?

model.train(dataset_train, dataset_val,
            learning_rate=config.LEARNING_RATE / 10,
            epochs=130,
            layers='all', augmentation = augmentation)

**Comment by hackerghost93** [comment]
If you are starting from coco weights definitely you need to train the heads for 20 epochs or so. In you case there is a sample called coco.py in the project. I recommend following there training scheme.
